src/routes/api/product.routes.js

```diff
@@ -23,4 +23,12 @@ productRoutes.post(
   uploadImages,
   createProduct
 );
+
+productRoutes.patch(
+  '/:productId',
+  protectRoute,
+  restrictTo('seller','admin'),
+  removeExpiredProducts,
+);
```

Comment:
- Here you need middleware to check if the product exists
- and middleware to check if the seller is the owner of the product